handlers/handlers.go

```diff
@@ -1,238 +1,238 @@
-
-	case http.MethodDelete:
-		// Cookie will overwrite existing cookie then delete itself
-		cookie := http.Cookie{
-			Name:     cerealNotesCookieName,
-			Value:    "",
-			HttpOnly: true,
-			MaxAge:   -1,
-		}
-
-		http.SetCookie(responseWriter, &cookie)
-		responseWriter.WriteHeader(http.StatusOK)
-		responseWriter.Write([]byte(fmt.Sprint("user succefully logged out")))
-	default:
-		respondWithMethodNotAllowed(
-			responseWriter,
-			[]string{http.MethodPost, http.MethodDelete})
-	}
-}
-
-type AuthentictedRequestHandlerType func(
-	http.ResponseWriter,
-	*http.Request,
-	models.UserId)
-
-func AuthenticateOrRedirectToLogin(
-	authenticatedHandlerFunc AuthentictedRequestHandlerType,
-) func(http.ResponseWriter, *http.Request) {
-	return func(responseWriter http.ResponseWriter, request *http.Request) {
-			if userId, err := getUserIdFromJwtToken(request); err != nil {
-				// If not loggedin redirect to login page
-				http.Redirect(
-					responseWriter,
-					request,
-					paths.LoginOrSignupPath,
-					http.StatusTemporaryRedirect)
-			} else {
-				authenticatedHandlerFunc(responseWriter, request, userId)
-			}
-		}
-}
-
-// Authenticated Handlers
-func HandleHomeRequest(
-	responseWriter http.ResponseWriter,
-	request *http.Request,
-	userId models.UserId,
-) {
-	switch request.Method {
-	case http.MethodGet:
-		parsedTemplate, err := template.ParseFiles("templates/home.tmpl")
-		if err != nil {
-			log.Fatal(err)
-		}
-
-		parsedTemplate.Execute(responseWriter, userId)
-	default:
-		respondWithMethodNotAllowed(responseWriter, []string{http.MethodGet})
-	}
-}
-
-// UTIL
-func respondWithMethodNotAllowed(
-	responseWriter http.ResponseWriter,
-	allowedMethods []string,
-) {
-	allowedMethodsString := strings.Join(allowedMethods, ", ")
-	responseWriter.Header().Set("Allow", allowedMethodsString)
-
-	statusCode := http.StatusMethodNotAllowed
-	http.Error(responseWriter, http.StatusText(statusCode), statusCode)
-}
-
-func getRequestBody(request *http.Request) []byte {
-	body, err := ioutil.ReadAll(request.Body)
-	if err != nil {
-		panic(err)
-	}
-
-	if err := request.Body.Close(); err != nil {
-		panic(err)
-	}
-
-	return body
-}
+package handlers 
+ 
+import ( 
+	"encoding/json" 
+	"fmt" 
+	"github.com/atmiguel/cerealnotes/models" 
+	"github.com/atmiguel/cerealnotes/paths" 
+	"github.com/atmiguel/cerealnotes/services/userservice" 
+	"github.com/dgrijalva/jwt-go" 
+	"html/template" 
+	"io/ioutil" 
+	"log" 
+	"net/http" 
+	"strings" 
+	"time" 
+) 
+ 
+const oneWeekInMinutes = 60 * 24 * 7 
+const credentialTimeoutInMinutes = oneWeekInMinutes 
+const cerealNotesCookieName = "CerealNotesToken" 
+ 
+type JwtTokenClaim struct { 
+	models.UserId `json:"userId"` 
+	jwt.StandardClaims 
+} 
+ 
+var tokenSigningKey []byte 
+ 
+func SetTokenSigningKey(key []byte) { 
+	tokenSigningKey = key 
+} 
+ 
+// HANDLERS 
+func HandleLoginOrSignupRequest( 
+	responseWriter http.ResponseWriter, 
```

Comment:
In general I completely agree with you guys that single letter variable names are a cancer, but I do use the signature `(w http.ResponseWriter, r  *http.Request)` a lot. This function signature is _super_ common. This is the rare exception where I think spelling out responseWriter and request instead of w and r made the code harder to read for me. Just because i'm so used to seeing `(w http.ResponseWriter, r  *http.Request)`, when I see that signature I immediately pattern match and think "oh, it's one of _those_ types of functions".

Reply:
Totally understand that, but i had a really hard time reading all the code i found written like that. Like damn one of the variables is `r` and i was always like wait is this the request or the response. I understand what you mean about getting used to things, but I often will `index` over `i` in for loops, as well as `event` over `e` in js event handlers. Just makes more sense to me